File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/users/UsersDeletePhotoRequest.java
package com.github.seratch.jslack.api.methods.request.users;

import com.github.seratch.jslack.api.methods.SlackApiRequest;

/**
 * Delete the user profile photo
 */
public class UsersDeletePhotoRequest implements SlackApiRequest {

    /**
     * Authentication token. Requires scope: `users.profile:write`
     */
    private String token;

    UsersDeletePhotoRequest(String token) {
        this.token = token;
    }

    public static UsersDeletePhotoRequestBuilder builder() {
        return new UsersDeletePhotoRequestBuilder();
    }

    public String getToken() {
        return this.token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public static class UsersDeletePhotoRequestBuilder {
        private String token;

        UsersDeletePhotoRequestBuilder() {
        }

        public UsersDeletePhotoRequest.UsersDeletePhotoRequestBuilder token(String token) {
            this.token = token;
            return this;
        }

        public UsersDeletePhotoRequest build() {
            return new UsersDeletePhotoRequest(token);
        }

        public String toString() {
            return "UsersDeletePhotoRequest.UsersDeletePhotoRequestBuilder(token=" + this.token + ")";
        }
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/api/methods/request/users/UsersDeletePhotoRequest.kt
package work.socialhub.kslack.api.methods.request.users

import work.socialhub.kslack.api.methods.FormRequest
import work.socialhub.kslack.api.methods.SlackApiRequest
import kotlin.js.JsExport

/**
 * Delete the user profile photo
 */
@JsExport
class UsersDeletePhotoRequest(
    /** Authentication token. Requires scope: `users.profile:write` */
    override var token: String?
) : SlackApiRequest, FormRequest{
    override fun toMap(): Map<String, Any> {
        return mapOf()
    }
}